src/SignIn.tsx

```diff
@@ -108,26 +109,103 @@ function SignInWithSeed({
   );
 }
 
-export function SignInFullScreen(): JSX.Element {
-  const path = useLocation();
+type LocationState = {
+  referrer?: string;
+};
+
+function rewriteIDs(event: UnsignedEvent): UnsignedEvent {
+  // TODO: This feels quite dangerous
+  const replacedTags = event.tags.map((tag) =>
+    tag.map((t) => t.replaceAll(UNAUTHENTICATED_USER_PK, event.pubkey))
+  );
+  return {
+    ...event,
+    content: event.content.replaceAll(UNAUTHENTICATED_USER_PK, event.pubkey),
+    tags: replacedTags,
+  };
+}
+
+function planRewriteUnpublishedEvents(
+  plan: Plan,
+  events: List<UnsignedEvent>
+): Plan {
+  const allEvents = plan.publishEvents.concat(events);
+  const rewrittenEvents = allEvents.map((event) =>
+    rewriteIDs({
+      ...event,
+      pubkey: plan.user.publicKey,
+    })
+  );
+  return {
+    ...plan,
+    publishEvents: rewrittenEvents,
+  };
+}
+
+export function SignInModal(): JSX.Element {
   const login = useLogin();
   const navigate = useNavigate();
-  document.body.classList.add("background");
+  const location = useLocation();
+  const { publishEventsStatus } = useData();
+  const { relayPool, finalizeEvent } = useApis();
+  const { createPlan, setPublishEvents } = usePlanner();
+  const referrer =
+    (location.state as LocationState | undefined)?.referrer || "/";
+  const onHide = (): void => {
+    navigate(referrer);
+  };
+  const setPrivateKey = async (pk: string): Promise<void> => {
+    const user = login(pk);
+    const plan = planRewriteUnpublishedEvents(
+      { ...createPlan(), user },
+      publishEventsStatus.unsignedEvents
+    );
+    if (plan.publishEventsStatus.unsignedEvents.size === 0) {
+      onHide();
+      return;
+    }
+    const results = await execute({
+      plan,
+      relayPool,
+      relays: plan.relays.filter((r) => r.write === true),
+      finalizeEvent,
+    });
+    setPublishEvents(() => {
+      return {
+        unsignedEvents: plan.publishEvents, // TODO: or better to empty it?
```

Comment:
i appended them in `executePlan`; since resending can be done again and again with more and more events; here, resending should also be possible after a login, so keep it like that